compact.go

```diff
@@ -77,48 +83,58 @@ func (c Compact) Bytes() []byte {
 	return append([]byte{(topSixBits << 2) + 3}, b...)
 }
 
-func DecodeCompact(buffer *bytes.Buffer) (Compact, error) {
+func DecodeCompact[T BigNumbers](buffer *bytes.Buffer) (Compact[T], error) {
 	decoder := Decoder{Reader: buffer}
 	result := make([]byte, 16)
 	b, err := decoder.DecodeByte()
 	if err != nil {
-		return Compact{}, err
+		return Compact[T]{}, err
 	}
 	mode := b & 3
 	switch mode {
 	case 0:
-		return Compact(NewU128(big.NewInt(0).SetUint64(uint64(b >> 2)))), nil
+		value := NewU128(uint64(b >> 2))
+		result := interface{}(value).(T)
```

Comment:
I think it will be cleaner if you implement `Interface()` method for each type (U8, U16 ...)
```go
func (n U8) Interface() BigNumbers {
	return n
}
```
and call `.Interface().(T)`

but you need to know the type T 
```go
switch reflect.Zero(reflect.TypeOf(*new(T))).Interface().(type) {
	case U8:
		switch v := v.(type) {
		case uint8:
			result := U8(value).Interface().(T)
		}
                ...
	}
        ...
```

Reply:
`Interface` is unnecessary. I suggested
```go
result := interface{}(value).(T)
```
as something that's working, but to be written in a better way. You can even do the following:
```go
Numeric(value).(T)
```